axelor-sale/src/main/resources/views/SaleOrderLine.xml

```diff
@@ -964,6 +963,12 @@
       if="eval: __user__.group?.isClient" expr="true"/>
   </action-attrs>
 
+  <action-attrs name="action-sale-order-line-attrs-hidden-panel">
+    <attribute name="hidden"
+      for="stockMoveLineOfSOPanel,projectTaskListPanel,invoicingFollowUpPanel" expr="true"
+      if="__parent__.statusSelect == 1 || __parent__.statusSelect == 2"/>
```

Comment:
Use static variable instead of static values. Also need to specify this action on onNew.